src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectionTestWithSSLCert/CertificateTestWithTdsServer.cs

```diff
@@ -0,0 +1,146 @@
+            string userId = string.IsNullOrWhiteSpace(builder.UserID) ? "user" : builder.UserID;
+            string password = string.IsNullOrWhiteSpace(builder.Password) ? "password" : builder.Password;
+
+            using TestTdsServer server = TestTdsServer.StartTestServer(enableFedAuth: false, enableLog: false, connectionTimeout: 15,
+                methodName: "", new X509Certificate2(s_fullPathToPfx, "nopassword", X509KeyStorageFlags.UserKeySet),
+                encryptionType: connectionTestParameters.TdsEncryptionType);
+
+            builder = new(server.ConnectionString)
+            {
+                UserID = userId,
+                Password = password,
+                TrustServerCertificate = connectionTestParameters.TrustServerCertificate,
+                Encrypt = connectionTestParameters.Encrypt,
+            };
+
+            if (connectionTestParameters.Certificate != string.Empty)
+            {
+                builder.ServerCertificate = connectionTestParameters.Certificate;
+            }
+
+            if (connectionTestParameters.HostNameInCertificate != string.Empty)
+            {
+                builder.HostNameInCertificate = connectionTestParameters.HostNameInCertificate;
+            }
+
+            using SqlConnection connection = new(builder.ConnectionString);
+            try
+            {
+                connection.Open();
+                Assert.Equal(connectionTestParameters.TestResult, (connection.State == ConnectionState.Open));
+            }
+            catch (Exception)
+            {
+                Assert.False(connectionTestParameters.TestResult);
+            }
+        }
+
+        private static void CreatePfxCertificate(string script)
+        {
+            string currentDirectory = Directory.GetCurrentDirectory();
+            string powerShellCommand = "powershell.exe";
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                powerShellCommand = "pwsh";
+            }
+
+            if (File.Exists(script))
+            {
+                StringBuilder output = new();
+                Process proc = new()
+                {
+                    StartInfo =
+                    {
+                        FileName = powerShellCommand,
+                        RedirectStandardError = true,
+                        RedirectStandardOutput = true,
+                        UseShellExecute = false,
+                        Arguments = $"{script} -OutDir {currentDirectory} > result.txt",
+                        CreateNoWindow = false,
+                        Verb = "runas"
+                    }
+                };
+
+                proc.EnableRaisingEvents = true;
+
+                proc.OutputDataReceived += new DataReceivedEventHandler((sender, e) =>
+                {
+                    output.AppendLine(e.Data);
```

Comment:
NIT: e.Data could be null. To add more `DataReceivedEventHandler` is called each time the process writes a line to the redirected StandardOutput stream. When the stream is closed `DataReceivedEventHandler` will be invoked one last time with 3.Data set to null. To avoid appending null values it is better to check for null. 
Something similar to below:
```C#
if (e.Data != null)
{
    output.AppendLine(e.Data);
}
```

Reply:
Condition to test for null value was implemented as suggested.